library/std/src/os/net/linux_ext/tcp.rs

```diff
@@ -98,6 +98,47 @@ pub trait TcpStreamExt: Sealed {
     #[unstable(feature = "tcp_deferaccept", issue = "119639")]
     #[cfg(target_os = "linux")]
     fn deferaccept(&self) -> io::Result<u32>;
+
+    /// Set the number of `SYN` packets to send before giving up establishing a connection.
+    ///
+    /// In case the server does not repond, a `SYN` packet is sent by the client.
+    /// This option controls the number of attempts, the default system value
+    /// can be seen via the `net.ipv4.tcp_syn_retries` sysctl's OID (usually 5 or 6).
+    /// The maximum valid value is 255.
+    ///
+    /// See [`man 7 tcp`](https://man7.org/linux/man-pages/man7/tcp.7.html)
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(tcp_syncnt)]
+    /// use std::net::TcpStream;
+    /// use std::os::linux::net::TcpStreamExt;
+    ///
+    /// let stream = TcpStream::connect("127.0.0.1:8080")
+    ///         .expect("Couldn't connect to the server...");
+    /// stream.set_syncnt(3).expect("set_setcnt call failed");
+    #[unstable(feature = "tcp_syncnt", issue = "123112")]
+    fn set_syncnt(&self, count: u32) -> io::Result<()>;
```

Comment:
Just out of curiosity - could the count parameter just be a u8 since the docs state a maximum valid value of 255 (or is it expected that this maximum might rise in the future)?

Reply:
it definitively can be u8.